libraries/shared/logs/extractor.go

```diff
@@ -56,26 +72,42 @@ func (extractor *LogExtractor) AddTransformerConfig(config transformer.EventTran
 		return checkedHeadersErr
 	}
 
-	if extractor.StartingBlock == nil {
-		extractor.StartingBlock = &config.StartingBlockNumber
-	} else if earlierStartingBlockNumber(config.StartingBlockNumber, *extractor.StartingBlock) {
+	if resetStartingBlockNumber(config.StartingBlockNumber, extractor.StartingBlock) {
 		extractor.StartingBlock = &config.StartingBlockNumber
 	}
 
+	if resetEndingBlockNumber(config.EndingBlockNumber, extractor.EndingBlock) {
+		extractor.EndingBlock = &config.EndingBlockNumber
+	}
+
 	addresses := transformer.HexStringsToAddresses(config.ContractAddresses)
 	extractor.Addresses = append(extractor.Addresses, addresses...)
 	extractor.Topics = append(extractor.Topics, common.HexToHash(config.Topic))
 	return nil
 }
 
+func resetStartingBlockNumber(currentTransformerBlock int64, extractorBlock *int64) bool {
+	return extractorBlock == nil || currentTransformerBlock < *extractorBlock
+}
+
+func resetEndingBlockNumber(currentTransformerBlock int64, extractorBlock *int64) bool {
+	if extractorBlock == nil {
+		return true
+	} else if currentTransformerBlock == int64(-1) {
+		return true
+	}
+
+	return *extractorBlock != int64(-1) && currentTransformerBlock > *extractorBlock
```

Comment:
Oh interesting, so you're thinking, that if we have a command to reset `check_count` to zero, then `execute` will automatically eventually pick up that header to fetch logs for, transform, etc? Great idea! I'll dig into that a bit further.

Reply:
Just pushed up some changes to remove the `transformEventsOnDemand` command in favor of a `resetHeaderCheckCount` command. Left some of the refactoring that was done perviously, like creating `NewLogDelegator` and `NewLogExtractor` method and passing those both into `NewEventWatcher` just in case we want to make some config changes to those in the future. But would be happy to remove those changes as well.